### ISSUE TITLE
bm/ovirt: pass API_VIP/DNS_VIP from cluster config/installer and render templates with it

### ISSUE THREAD

**Issue by rgolangh** [openshift/machine-config-operator#812]
I'm developing the platform provider for ovirt #766, which is a platform which doesn't provide a DNS service, and as such it utilzes the work done on metal3. To bootstrap a working DNS solution, the installer asks the user for an IP to be the virtual IP of the DNS - DNS_VIP.
This DNS_VIP is a value which must end up in the masters ignition files for the ovirt platform, and possibly others. 

Today I see that there are number of files which bootstrap MCO, one of them is `config-file`:
https://github.com/openshift/machine-config-operator/blob/acf5b49de3692801ce9cd3d799f64961f9283b00/cmd/machine-config-operator/bootstrap.go#L74

Which is the serialized `InstallConfig` of the installer, which has all I need:
https://github.com/openshift/installer/blob/479ca0fc0fd46c489c4b97d783bf1dd58cc373a2/pkg/types/installconfig.go#L42

What I want is to add this ClusterConfigMap to the RenderConfig and by that I could render my templates.
Can I rely on the installer InstallConfig type and to serialize the content to it and put it in render config? It creates dependency on the installer types, and I don't know yet how does it fit the overall design of those components.

**Comment by kikisdeliveryservice** [comment]
Briefly looking through, it looks like the following prs would need to land before we'd tackle this:
https://github.com/openshift/api/pull/314
https://github.com/openshift/machine-config-operator/pull/767

What is the status of the first PR in openshift/api ??

**Comment by rgolangh** [comment]
I have nothing to change there, just waiting for approval

**Comment by kikisdeliveryservice** [comment]
specifically which value are you needing from the cluster config?

**Comment by kikisdeliveryservice** [comment]
Also if you can add more details about the platform, that would be great as I don't think we have that much familiarity with it.

**Comment by rgolangh** [comment]
You'll find it here, under the new 'ovirt' platform (not merged yet obviously) 
https://github.com/rgolangh/installer/blob/fd85cbdb57e9d3e5ba20bba757468c9b8c57642d/pkg/types/ovirt/platform.go#L27

**Comment by rgolangh** [comment]
I can think of more data which I need to share, like the the platform authentication details, which will be used by the cluster-api or actuator to provision machines

**Comment by kikisdeliveryservice** [comment]
can you outline what PRs need to merge / the overall status of this ovirt platform? before this PR could even merge?

**Comment by runcom** [comment]
> What I want is to add this ClusterConfigMap to the RenderConfig and by that I could render my templates.

we're _Rendering_ using the cluster config already - what's the reason to be in RenderConfig as well? you might want it in ControllerConfig maybe (still looking at the code and the related ovirt PRs)

**Comment by rgolangh** [comment]
RenderConfig is the one that is used to execute templating on templates/master/... no?

**Comment by kikisdeliveryservice** [comment]
It seems that this PR would be a bit away since a bunch of other PRs need to actually merge. For ex: first need to get that installer config PR merged before the MCO can do anything with your new data type.

**Comment by abhinavdahiya** [comment]
Installer team is deprecating all clients of the `InstallConfig` in the cluster ie. `ClusterConfig` `ConfigMap`. Please move to `infrastructures.config.openshift.io`.

**Comment by runcom** [comment]
> Please move to `infrastructures.config.openshift.io`.

cool, we do play with that already as well in code (iirc, can't check right now).

**Comment by rgolangh** [comment]
@kikisdeliveryservice speculative? no. Its a work in progress toward making it happen.
For the installer to support ovirt, or almost any platform, I must to have MCO support, its integral, but yes the patch to at least getting it into the api platform type is the start.

@runcom got any reference to infrastructures.config.openshift.io?

**Comment by bcrochet** [comment]
I'd like to add that the Baremetal platform also needs some/all of these values as well. That is more pressing as we are trying to get Baremetal support into 4.2. I currently have PR #795  that adds some static pod definitions that would need at a minimum the API vip and the DNS vip.

**Comment by runcom** [comment]
> I currently have PR #795 that adds some static pod definitions that would need at a minimum the API vip and the DNS vip. Should I open a separate issue for that?

if it's tackling the same issue, let's generalize this issue.

**Comment by rgolangh** [comment]
> RenderConfig is the one that is used to execute templating on templates/master/... no?

I see that the controller bootstrap does that:
cmd/machine-config-controller/bootstrap.go#runbootstrapCmd

So this means I won't  have access the cluster config

**Comment by runcom** [comment]
> @runcom got any reference to infrastructures.config.openshift.io?

@rgolangh we're playing with that in pkg/operator but I'm not sure how that's handled outside the MCO to populate it:

```
	infra, err := optr.infraLister.Get("cluster")
	if err != nil {
		return nil, nil, err
	}
```

I believe it's setup in the installer :thinking:

**Comment by rgolangh** [comment]
> Installer team is deprecating all clients of the `InstallConfig` in the cluster ie. `ClusterConfig` `ConfigMap`. Please move to `infrastructures.config.openshift.io`.

@abhinavdahiya  I don't know what this namespace refer to - I saw you mentioning this several times - can you please explain this?
If I get you correctly and following @runcom comment, If I want to shar edata from the installer then I should exapand `InfrastructureStatus` struct?

**Comment by celebdor** [comment]
Issue explaining the move of installconfig stuff to infrastructures.config.openshift.io https://github.com/openshift/installer/issues/680

**Comment by celebdor** [comment]
I think it's about adding to https://github.com/openshift/api/blob/master/config/v1/types_infrastructure.go#L99

**Comment by rgolangh** [comment]
@runcom I think we can link #943 with this issue and edit its name - what do you think?
